test/config/utility.cc

```diff
@@ -348,14 +368,6 @@ ConfigHelper::ConfigHelper(const Network::Address::IpVersion version, Api::Api&
 
   for (int i = 0; i < static_resources->clusters_size(); ++i) {
     auto* cluster = static_resources->mutable_clusters(i);
-    if (!cluster->hosts().empty()) {
```

Comment:
Could you assert here and below that the deprecated hosts are empty just to avoid any confusing oopses?

Reply:
Given it's deprecated, shouldn't we go with the normal way where we let folks deprecate when it goes fatal, or are we worried about mixing and matching?  It's nice to give folks the grace period to migrate config and tests over.

Reply:
I found it pretty confusing to have both around IMHO and we should try and force folks to move to `load_assignment`, at least for the integration tests. Part of this goes back to the original design of how we populate host ports, by doing an ordered traversal of the hosts. Some tests then want to go in and do a bunch of surgery on the load assignment, but then need to worry about clearing out hosts, which is totally unrelated, and vice versa. This is not the most declarative thing to have.

I would like to eventually move away from this model, perhaps towards registering configuration lambdas that can mutate specific entries with the late port binding, but that's a cleanup PR.

Reply:
Got it.  I guess we can special case if it's painful to support both, it's just kinda harsh to make everyone update their tests when they pull HEAD.  This is a pretty special case though so your call!